### PR TITLE
add findAll(GET) method in the campaign-application.service(second try)

### PR DESCRIPTION
**Pull request by Martbul**
Closes #{issue number}

### New environment variables:

- [ ] `NEW_ENV_VAR`: env var details

### New or updated dependencies:

Dependency name|Previous version|Updated version|Details
---|---|---|---
`dependency/name`|`v1.0.0`|`v2.0.0`|

### CODE REVIEW

File: apps/api/src/campaign-application/campaign-application.service.spec.ts [hunk 1]
```diff
@@ -2,14 +2,57 @@ import { Test, TestingModule } from '@nestjs/testing'
...
-      providers: [CampaignApplicationService],
-    }).compile()
+      providers: [
+        {
+          // Use the interface as token
+          provide: NotificationsProviderInterface,
+          // But actually provide the service that implements the interface
+          useClass: SendGridNotificationsProvider,
+        },
+        CampaignService,
```
Comment (gparlakov) [inline apps/api/src/campaign-application/campaign-application.service.spec.ts:36]: Do we need all these providers? CampaignService? Vault Service.... and all the rest? 
It looks like `CampaignApplicationService` is the only requirement for this PR's scope

Comment (gparlakov) [reply apps/api/src/campaign-application/campaign-application.service.spec.ts:36]: Keep in mind that all of these would need to be instantiated with all of their dependencies for each test!!! 

Instead use mocking (see other tests or look for autoSpy usage)

File: apps/api/src/campaign-application/campaign-application.service.ts [hunk 2]
```diff
@@ -20,7 +22,7 @@ export class CampaignApplicationService {
   }
 
   findAll() {
-    return `This action returns all campaignApplication`
+    return this.prisma.campaignApplication.findMany()
```
Comment (gparlakov) [inline apps/api/src/campaign-application/campaign-application.service.ts:25]: The controller would need to be updated to check for authenticated user and allow getting only for admin/operator roles.

**Review by gparlakov — COMMENTED** [review]
needs a bit more ❤️

File: apps/api/src/campaign-application/campaign-application.service.spec.ts
Comment (gparlakov) [inline apps/api/src/campaign-application/campaign-application.service.spec.ts:1]: Please remove commented out code

**Review by gparlakov — COMMENTED** [review]
Thanks for updating.

File: apps/api/src/campaign-application/campaign-application.service.spec.ts
Comment (Martbul) [reply apps/api/src/campaign-application/campaign-application.service.spec.ts:1]: Sure. I have missed it.

**Comment by Martbul** [comment]
Трябва ли да правя някакви промени?

**Pull request merged**